src/psyclone/tests/psyir/nodes/assignment_test.py

```diff
@@ -338,3 +338,32 @@ def test_assignment_gen_code():
     assignment.gen_code(module)
     code = str(module.root)
     assert "tmp = 0.0\n" in code
+
+
+def test_pointer_assignment():
+    ''' Test that pointer assingments work as expected '''
+    lhs = Reference(Symbol("var1"))
+    rhs = Reference(Symbol("var2"))
+
+    # Constructors and creators
+    assignment1 = Assignment(is_pointer=True)
+    assignment1.addchild(lhs.copy())
+    assignment1.addchild(rhs.copy())
+    assignment2 = Assignment.create(lhs, rhs, is_pointer=True)
+    not_pointer = Assignment.create(lhs.copy(), rhs.copy())
+
+    # Getters and equality
+    assert assignment1.is_pointer
+    assert assignment2.is_pointer
+    assert not not_pointer.is_pointer
+    assert assignment1 == assignment2
+    assert assignment1 != not_pointer
+
+    # Str and setter
+    assert "Assignment[is_pointer=True]" in str(assignment1)
+    with pytest.raises(TypeError) as err:
+        assignment1.is_pointer = None
+    assert ("is_pointer must be a boolean but got 'NoneType'"
+            in str(err.value))
+    assignment1.is_pointer = False
+    assert "Assignment[]" in str(assignment1)
```

Comment:
Please could you also check that copying an Assignment node preserves the `is_pointer` attribute.

Reply:
Done